src/convertToObject.js

```diff
@@ -3,16 +3,28 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS
+ * properties (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const newObjectwithClasses = sourceString
+    .split(';')
+    .map(classes => classes.trim())
+    .filter(Boolean)
+    .map(classes => classes.split(':'))
+    .reduce((acc, current) => ({
+      ...acc,
+      [current[0].trim()]: current[1].trim(),
+    }), {});
```

Comment:
Not Bad)
But our mentors want that you do all this operations in reduce method by 1 iteration :)